### PR TITLE
Check validity of identity templates

### PR DESCRIPTION
**Pull request by kaniyan**

### CODE REVIEW

File: services/device-registry-base/src/main/java/org/eclipse/hono/service/management/device/IdentityTemplate.java [hunk 1]
```diff
@@ -135,6 +138,28 @@ public String apply(final String subjectDN) {
...
+     * @throws NullPointerException if template is {@code null}.
+     * @throws IllegalArgumentException if the template does not contain any placeholders.
+     */
+    public static void checkValidity(final String template) {
+        if (!template.contains(RegistryManagementConstants.PLACEHOLDER_SUBJECT_DN)) {
+            Arrays.stream(Attribute.values())
+                    .map(Attribute::getPlaceHolder)
+                    .filter(template::contains)
+                    .findFirst()
+                    .orElseThrow(() -> new IllegalArgumentException("template must contain at least one placeholder"));
```
Comment (sophokles73) [inline services/device-registry-base/src/main/java/org/eclipse/hono/service/management/device/IdentityTemplate.java:154]: why does it need to contain at least one placeholder? However, we should probably be able to detect a case where an unsupported placeholder is being used, right?

Comment (kaniyan) [reply services/device-registry-base/src/main/java/org/eclipse/hono/service/management/device/IdentityTemplate.java:154]: As per the spec,
> The template MUST contain at least one of these placeholders.

We already do this check inside `TrustedCertificateAuthority`. I have refactored it now to check for attributes dynamically.

Comment (sophokles73) [reply services/device-registry-base/src/main/java/org/eclipse/hono/service/management/device/IdentityTemplate.java:156]: What about checking for unsupported placeholders?

Comment (kaniyan) [reply services/device-registry-base/src/main/java/org/eclipse/hono/service/management/device/IdentityTemplate.java]: I have added a check for unsupported placeholders. Would you mind taking another look?

**Review by sophokles73 — APPROVED** [review]
LGTM

**Pull request merged**